controllers/torqueBalancing/init/initVisualizer.m

```diff
@@ -21,9 +21,37 @@
 
 %% Robot simulator
 if CONFIG.visualize_robot_simulator == 1
+    % list of joints used in the visualizer
+    CONFIG.mdlLdr           = iDynTree.ModelLoader();
+    CONFIG.consideredJoints = iDynTree.StringVector();
     
-    references        = trajectoryGenerator(CONFIG.initForKinematics.xCoM,t,CONFIG);
-    CONFIG.figureCont = visualizeSimulation(t,chi,CONFIG,references);
+    CONFIG.consideredJoints.push_back('torso_pitch');
```

Comment:
This should be automatically extracted by the used joint lists, but at the moment that is not exposed in mex-wholebodymodel, so I see why you had to hardcode this.